optuna/terminator/erroreval.py

```diff
@@ -0,0 +1,59 @@
+from __future__ import annotations
+
+import abc
+
+import numpy as np
+
+from optuna._experimental import experimental_class
+from optuna.study.study import Study
+from optuna.trial import Trial
+from optuna.trial._state import TrialState
+
+
+_CROSS_VALIDATION_SCORES_KEY = "terminator:cv_scores"
+
+
+class BaseErrorEvaluator(metaclass=abc.ABCMeta):
+    @abc.abstractmethod
+    def evaluate(self, study: Study) -> float:
+        pass
+
+
+@experimental_class("3.2.0")
+class CrossValidationErrorEvaluator(BaseErrorEvaluator):
+    def evaluate(self, study: Study) -> float:
+        assert len(study.get_trials(states=(TrialState.COMPLETE,))) != 0
+
+        best_trial_attrs = study.best_trial.system_attrs
+        if _CROSS_VALIDATION_SCORES_KEY in best_trial_attrs:
+            cv_scores = best_trial_attrs[_CROSS_VALIDATION_SCORES_KEY]
+        else:
+            raise ValueError(
+                "Cross-validation scores have not been reported. Please call "
+                "`report_cross_validation_scores(trial, scores)` during a trial and pass the "
+                "list of scores as `scores`."
+            )
+
+        k = len(cv_scores)
+        scale = 1 / k + 1 / (k - 1)
```

Comment:
Might improve code readability addressing zero-division.
```suggestion
        k = len(cv_scores)
        assert k > 1, "Should be guaranteed by `report_cross_validation_scores`."
        scale = 1 / k + 1 / (k - 1)
```